### PR TITLE
Issue147 optional checksum

### LINKED ISSUE #147: checksum calculation should be made optional
Big files require a lot of time for checksum calculation. It is better to make this optional, and by default not to calculate the checksum.

### PR DESCRIPTION
**Pull request by johtoblan**
**Summary**:

Adds -c as a flag to nc2mmd.py script, to toggle if checksum should be calculated (Default is True)

**Related issue**: 

Closes #147 

**Suggested reviewer(s)**:

@mortenwh 
@ElodieFZ 

**Reviewer checklist**:

- [ ] The headers of all files contain a reference to the repository license (i.e., "License: This file is part of py-mmd-tools, licensed under the Apache License 2.0 (https://www.apache.org/licenses/LICENSE-2.0)")
- [ ] 100% test coverage of new code - meaning:
    - [ ] The overall test coverage increased or remained the same as before
    - [ ] Every function is accompanied with a test suite
    - [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
    - [ ] Functions with optional arguments have separate tests for all options
- [ ] Examples are supported by doctests
- [ ] All tests are passing
- [ ] All names (e.g., files, classes, functions, variables) are explicit
- [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: py_mmd_tools/nc_to_mmd.py [hunk 1]
```diff
@@ -988,18 +988,31 @@ def to_mmd(self, mmd_yaml=None, *args, **kwargs):
...
+            }
+
+        else:
+            self.metadata['storage_information'] = {
+                'file_name': os.path.basename(self.netcdf_product),
+                'file_location': file_location,
+                'file_format': 'NetCDF-CF',
+                'file_size': '%.2f'%file_size,
+                'file_size_unit': 'MB',
+            }
```
Comment (mortenwh) [inline py_mmd_tools/nc_to_mmd.py:1015]:
```suggestion
        self.metadata['storage_information'] = {
                'file_name': os.path.basename(self.netcdf_product),
                'file_location': file_location,
                'file_format': 'NetCDF-CF',
                'file_size': '%.2f'%file_size,
                'file_size_unit': 'MB',
            }
        if checksum_calculation:
            md5hasher = FileHash('md5', chunk_size=1048576)
            fchecksum = md5hasher.hash_file(file_for_checksum_calculation)

            self.metadata['storage_information']['checksum'] = fchecksum
            self.metadata['storage_information']['checksum_type'] = '%ssum' % md5hasher.hash_algorithm
```

Comment (mortenwh) [reply py_mmd_tools/nc_to_mmd.py:1015]: it's also a bit strange with `%ssum` in `self.metadata['storage_information']['checksum_type'] = '%ssum' % md5hasher.hash_algorithm` - should the `sum` part be removed?

File: tests/test_nc_to_mmd.py [hunk 2]
```diff
@@ -1079,13 +1079,22 @@ def test_checksum(self):
...
         with open(tested, 'w') as tt:
             tt.write('%s *%s'%(checksum, fn))
         md5hasher = FileHash('md5')
         self.assertTrue(md5hasher.verify_checksums(tested)[0].hashes_match)
 
+    def test_checksum_off(self):
+        """ToDo: Add docstring"""
+        fn = 'tests/data/reference_nc.nc'
+        nc2mmd = Nc_to_mmd(fn, check_only=True)
+        nc2mmd.to_mmd(checksum_calculation=False)
```
Comment (mortenwh) [inline tests/test_nc_to_mmd.py:1093]: should also be tested without input to ensure the default is correct..

**Review by mortenwh — CHANGES_REQUESTED** [review]
A few minor changes are needed...

File: tests/test_nc_to_mmd.py (hunk 2 above)
Comment (johtoblan) [reply tests/test_nc_to_mmd.py:1093]: will do

**Pull request merged**